Core/src/org/sleuthkit/autopsy/ingest/FileIngestPipeline.java

```diff
@@ -136,6 +140,14 @@ Date getStartTime() {
                     break;
                 }
             }
+            
+            // Save any properties that have not already been saved to the database
+            try{
+                file.save(Case.getCurrentCase().getSleuthkitCase());
+            } catch (TskCoreException ex){
+                Logger.getLogger(FileIngestPipeline.class.getName()).log(Level.SEVERE, "Failed to save data", ex); //NON-NLS
+            }
+            
```

Comment:
I think we should do two things here:

1. Move this code inside the job cancellation check conditional. We do not need to do a database round trip if nothing has changed.
2. Add the object id of the file to the error message to make debugging easier.